component/distributed_cache/distributed_cache.go

```diff
@@ -356,68 +361,265 @@ func (dc *DistributedCache) StreamDir(options internal.StreamDirOptions) ([]*int
 }
 
 func (dc *DistributedCache) CreateFile(options internal.CreateFileOptions) (*handlemap.Handle, error) {
+	var dcFile *fm.DcacheFile
+	var handle *handlemap.Handle
+	var err error
 	isAzurePath, isDcachePath, rawPath := getFS(options.Name)
-	var file *fm.DcacheFile
 	if isAzurePath {
 		log.Debug("DistributedCache::CreateFile : Path is having Azure subcomponent, path : %s", options.Name)
 		options.Name = rawPath
-		_, err := dc.NextComponent().CreateFile(options)
+		handle, err = dc.NextComponent().CreateFile(options)
 		if err != nil {
+			log.Err("DistributedCache::CreateFile : File Creation failed with err : %s Azure subcomponent, path : %s", err.Error(), options.Name)
 			return nil, err
 		}
 	} else if isDcachePath {
 		log.Debug("DistributedCache::CreateFile : Path is having Dcache subcomponent, path : %s", options.Name)
 		options.Name = rawPath
-		var err error
-		file, err = fm.NewDcacheFile(rawPath)
+		dcFile, err = fm.NewDcacheFile(rawPath)
 		if err != nil {
+			log.Err("DistributedCache::CreateFile : File Creation failed with err : %s Dcache subcomponent, path : %s", err.Error(), options.Name)
 			return nil, err
 		}
 	} else {
-		var err error
-		options.Name = rawPath
-		file, err = fm.NewDcacheFile(rawPath)
+		common.Assert(options.Name == rawPath)
+		dcFile, err = fm.NewDcacheFile(rawPath)
 		if err != nil {
+			log.Err("DistributedCache::CreateFile : File Creation failed with err : %s Dcache subcomponent, path : %s", err.Error(), options.Name)
 			return nil, err
 		}
 
-		_, err = dc.NextComponent().CreateFile(options)
+		handle, err = dc.NextComponent().CreateFile(options)
 		if err != nil {
+			log.Err("DistributedCache::CreateFile : File Creation failed with err : %s Azure subcomponent, path : %s", err.Error(), options.Name)
 			return nil, err
 		}
 		// todo : if one is success and other is failure, get to the previous state by removing the
 		// created entries for the files.
```

Comment:
Create should fail if either of the create fails. Since user didn't explicitly specify the namespace, our semantics is that we will create the file both in cache as well as azure, so failing to create in either should cause create to fail.